src/serverroutes.js

```diff
@@ -629,4 +633,65 @@ module.exports = function(app, saveSecurityConfig, getSecurityConfig) {
       })
     )
   })
+
+  app.post('/restore', (req, res, next) => {
+    const busboy = new Busboy({ headers: req.headers })
+    busboy.on('file', (fieldname, file, filename, encoding, mimetype) => {
+      try {
+        if (!filename.endsWith('.backup')) {
+          res.status(400).send('Invalid backup file')
```

Comment:
should the prefix `signalk-` be also checked? A more friendly error message about what's wrong with the file?

Reply:
ok